src/pybind/mgr/telemetry/module.py

```diff
@@ -0,0 +1,355 @@
+                    'min_size': pool['min_size'],
+                    'crush_rule': pool['crush_rule']
+                }
+            )
+
+        report['osd'] = {
+            'count': len(osd_map['osds']),
+            'require_osd_release': osd_map['require_osd_release'],
+            'require_min_compat_client': osd_map['require_min_compat_client']
+        }
+
+        report['fs'] = {
+            'count': len(fs_map['filesystems'])
+        }
+
+        report['metadata'] = dict()
+        report['metadata']['osd'] = self.gather_osd_metadata(osd_map)
+        report['metadata']['mon'] = self.gather_mon_metadata(mon_map)
+
+        report['usage'] = {
+            'pools': len(df['pools']),
+            'pg_num:': num_pg,
+            'total_objects': df['stats']['total_objects'],
+            'total_used_bytes': df['stats']['total_used_bytes'],
+            'total_bytes': df['stats']['total_bytes'],
+            'total_avail_bytes': df['stats']['total_avail_bytes']
+        }
+
+        return report
+
+    def send(self, report):
+        self.log.info('Upload report to: %s', self.config['url'])
+        proxies = dict()
+        if self.config['proxy']:
+            self.log.info('Using HTTP(S) proxy: %s', self.config['proxy'])
+            proxies['http'] = self.config['proxy']
+            proxies['https'] = self.config['proxy']
+
+        requests.put(url=self.config['url'], json=report, proxies=proxies)
+
+    def handle_command(self, command):
+        if command['prefix'] == 'telemetry config-show':
+            return 0, json.dumps(self.config), ''
+        elif command['prefix'] == 'telemetry config-set':
+            key = command['key']
+            value = command['value']
+            if not value:
+                return -errno.EINVAL, '', 'Value should not be empty or None'
+
+            self.log.debug('Setting configuration option %s to %s', key, value)
+            self.set_config_option(key, value)
+            self.set_config(key, value)
+            return 0, 'Configuration option {0} updated'.format(key), ''
+        elif command['prefix'] == 'telemetry send':
+            self.last_report = self.compile_report()
+            self.send(self.last_report)
+            return 0, 'Report send to {0}'.format(self.config['url']), ''
+        elif command['prefix'] == 'telemetry show':
+            report = self.last_report
+            if not report:
+                report = self.compile_report()
+            return 0, json.dumps(report), ''
+        else:
+            return (-errno.EINVAL, '',
+                    "Command not found '{0}'".format(command['prefix']))
+
+    def shutdown(self):
+        self.run = False
+        self.event.set()
+
+    def serve(self):
+        self.init_module_config()
+        self.run = True
+
+        self.log.debug('Waiting for mgr to warm up')
+        self.event.wait(10)
+
+        while self.run:
+            if not self.str_to_bool(self.config['enabled']):
```

Comment:
i don't think it's necessary to convert the value to boolean. as `self.config['enabled']` is already a boolean when being added to this dictionary.